plugins/broadcast/server/server.go

```diff
@@ -0,0 +1,165 @@
+	log          *logger.Logger
+	active       bool
+}
+
+var (
+	connectionList      = [256]Connection{}
+	connectionListMutex sync.RWMutex
+	index               atomic.Uint32
+)
+
+// Listen starts a TCP listener and starts a Connection for each accepted connection
+func Listen(bindAddress string, log *logger.Logger, shutdownSignal <-chan struct{}) error {
+
+	listener, err := net.Listen("tcp", bindAddress)
+	if err != nil {
+		return fmt.Errorf("failed to start Broadcast daemon: %w", err)
+	}
+
+	go func() {
+		for {
+			connection, err := listener.Accept()
+			if err != nil {
+				return
+			}
+			log.Infof("Started connection: %s", connection.RemoteAddr().String())
+			go handleConnection(connection, log, shutdownSignal)
+		}
+	}()
+
+	go func() {
+		defer func(listener net.Listener) {
+			err := listener.Close()
+			if err != nil {
+				log.Infof("Error closing listener: %s", err)
+			}
+		}(listener)
+
+		<-shutdownSignal
+
+		log.Infof("Closing Broadcast server...")
+		idx := int(index.Load())
+		connectionListMutex.Lock()
+		for i := 0; i < idx; i++ {
+			connectionList[i].active = false
+		}
+		connectionListMutex.Unlock()
+		log.Infof("Closing Broadcast server... \tDone")
+	}()
+
+	return nil
+}
+
+func handleConnection(conn net.Conn, log *logger.Logger, shutdownSignal <-chan struct{}) {
+	connectionListMutex.Lock()
+	defer connectionListMutex.Unlock()
+
+	idx := int(index.Load())
+	connectionList[idx] = Connection{
+		bufferedConn: buffconn.NewBufferedConnection(conn, tangle.MaxMessageSize),
+		log:          log,
+		active:       true,
+	}
+	index.Inc()
+
+	bufferedConnDataReceived, bufferedConnClosed := connectionList[idx-1].readLoop()
+
+	select {
+	case data := <-bufferedConnDataReceived:
+		//No input required. For debugging it will be printed
+		connectionList[idx-1].log.Infof("Data received:%v", data)
+		connectionList[idx-1].active = false
+	case <-shutdownSignal:
+		connectionList[idx-1].log.Infof("Shutdown signal received")
+		for i := 0; i < idx; i++ {
+			connectionList[i].active = false
+		}
+		return
+	case <-bufferedConnClosed:
+		connectionList[idx-1].log.Errorf("Connection lost")
+		connectionList[idx-1].active = false
+		return
+	}
+}
+
+func (connection *Connection) readLoop() (chan []byte, chan bool) {
+	bufferedConnDataReceived := make(chan []byte)
+	bufferedConnClosed := make(chan bool)
+
+	go func() {
+		{
+			cl := events.NewClosure(func() { close(bufferedConnClosed) })
+			connection.bufferedConn.Events.Close.Attach(cl)
+			defer connection.bufferedConn.Events.Close.Detach(cl)
+		}
+
+		{
+			cl := events.NewClosure(func(data []byte) {
+				d := make([]byte, len(data))
+				copy(d, data)
+				bufferedConnDataReceived <- d
+			})
+			connection.bufferedConn.Events.ReceiveMessage.Attach(cl)
+			defer connection.bufferedConn.Events.ReceiveMessage.Detach(cl)
+		}
+
+		if err := connection.bufferedConn.Read(); err != nil {
+			if err != io.EOF && !strings.Contains(err.Error(), "Use of closed network connection") {
```

Comment:
Instead of checking the string of the error, use `!errors.Is(nestedErr, ErrClosed)` where `ErrClosed` is from the `net` package.